### PR TITLE
Feat/https

### PR DESCRIPTION
**Pull request by viv-4**
Generates and uses self signed SSL certs on install (unless they already exist)
Cert will be for `PLACE_DOMAIN` and valid for 365 days.

Open to any ideas for changes to the process

### CODE REVIEW

File: config/nginx/nginx.conf [hunk 1]
```diff
@@ -32,69 +37,52 @@ http {
...
-      proxy_pass  http://api;
+    location /auth/ {
+      proxy_pass  http://auth;
       proxy_redirect   off;
       proxy_http_version 1.1;
-      proxy_set_header Host $host;
+      proxy_set_header Host $http_host;
       proxy_set_header X-Real-IP $remote_addr;
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-      proxy_set_header X-Forwarded-Proto https;
```
Comment (stakach) [inline config/nginx/nginx.conf:83]: shouldn't this be HTTPS now? I think it was wrong previously
it's used as a flag to indicate if HTTPS is being used and has been terminated at nginx - this is now indicating that nginx is using HTTP which might actually cause more issues as it'll attempt to redirect to HTTPS

File: docker-compose.yml [hunk 2]
```diff
@@ -180,6 +180,7 @@ services:
       PLACE_USERNAME: ${PLACE_USERNAME:-Place Support (localhost:8080)}
```
Comment (caspiano) [inline docker-compose.yml:180]:
```suggestion
      PLACE_USERNAME: ${PLACE_USERNAME:-Place Support (localhost:8443)}
```

File: install [hunk 3]
```diff
@@ -49,10 +49,16 @@ do
   esac
 done
 
+source .env
 PLACE_DOMAIN=${PLACE_DOMAIN:="localhost:8080"}
```
Comment (caspiano) [inline install:53]:
```suggestion
PLACE_DOMAIN=${PLACE_DOMAIN:="localhost:8443"}
```

File: install [hunk 4]
```diff
@@ -49,10 +49,16 @@ do
...
 PLACE_EMAIL=${PLACE_EMAIL:="support@place.tech"}
 PLACE_PASSWORD=${PLACE_PASSWORD:="development"}
 
+if [ ! -f ./ssl/default-domain/fullchain.pem ]; then
+  echo "=== Generating SSL certificates..."
+  ./scripts/generate-ssl ${PLACE_DOMAIN} 2> /dev/null || (printf "\nfailed to generate SSL certificates\n" && exit 1)
+fi
+
```
Comment (caspiano) [inline install:61]: Since this is idempotent, may as well throw it into the `restart` script as well.
Ideally we should only have one script.

**Pull request merged**